frontend/src/components/AutoLogin.jsx

```diff
@@ -0,0 +1,39 @@
+import React from "react";
+import { connect } from "react-redux";
+
+import proxy from "../proxy.js";
+
+class UnconnectedAutoLogin extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {};
+  }
+  componentDidMount = () => {
+    if (this.props.loggedIn) {
+      return;
+    }
+    fetch(proxy + "/auto-login", {
+      method: "POST",
+      credentials: "include"
+    })
+      .then(response => {
+        return response.text();
+      })
+      .then(resBody => {
+        let parsedBody = JSON.parse(resBody);
+        if (parsedBody.success) {
+          this.props.dispatch({ type: "login" });
+        }
+      });
+  };
+  render = () => {
+    return <div />;
```

Comment:
By convention*, components that don't render UI typically `return null`. 

*fun fact, I kinda helped cement this convention, in a [very embarrassing 2017 talk](https://www.youtube.com/watch?v=byCfsEP2gPk) I gave on the subject